### ISSUE TITLE
Allow users to attach several leads to a single contract

### ISSUE THREAD

**Issue by daamien** [FHenry/lead#14]
We'd like to attach multiple leads to the same contract.

**Comment by FHenry** [comment]
https://github.com/Dolibarr/dolibarr/pull/4144

**Comment by FHenry** [comment]
Done in module version 1.8 (branch master)

**Comment by daamien** [comment]
FTR when I activate the option here /custom/lead/admin/admin_lead.php ,  I get the following errors in the PostgreSQL logs :  

```
2015-12-04 09:42:45 CET [27009-1] labo@laboerp ERROR:  relation "idx_llx_lead_fk_soc" already exists
2015-12-04 09:42:45 CET [27009-2] labo@laboerp STATEMENT:  -- ALTER TABLE llx_lead ADD INDEX idx_llx_lead_fk_soc (fk_soc) replaced by --
        CREATE INDEX idx_llx_lead_fk_soc  ON llx_lead  (fk_soc)
2015-12-04 09:42:45 CET [27009-3] labo@laboerp ERROR:  constraint "llx_lead_ibfk_3" for relation "llx_lead" already exists
2015-12-04 09:42:45 CET [27009-4] labo@laboerp STATEMENT:  ALTER TABLE llx_lead ADD CONSTRAINT llx_lead_ibfk_3 FOREIGN KEY (fk_soc) REFERENCES llx_societe (rowid) DEFERRABLE INITIALLY IMMEDIATE;
```

However the feature works as expected

**Comment by daamien** [comment]
Found a tiny issue: 

The features works fine on /custom/lead/lead/card.php : we're able to link a lead to a contract which is already link to another lead.

However if you do it the other way, it's not possible. If you go to /contrat/card.php and if the contract is already linked to a lead, you can't add another lead.

**Comment by FHenry** [comment]
First comment "PostgreSQL logs", yes I know, but I don't if there is "ADD INDEX IF NOT EXISTS" stuff.
It will be the same thing each time you activate a module that declare index I think.

**Comment by FHenry** [comment]
second commet FIX in 1.9 (branch master)

**Comment by daamien** [comment]
works fine

merci :)
